### PR TITLE
[Testing:Submission] Stabilize late submission Cypress

### LINKED ISSUE #12732: Cypress late_submission_warning_messages failing on main
**Describe the bug**
The late_submission_warning_messages.spec.js Cypress test has 4 failures on main. The tests expect a different number of late days than what the application returns. For example, the test expects "Your submission will be 4 day(s) late" but the application returns "Your submission will be 3 day(s) late."

**Expected behavior**
All 11 tests in late_submission_warning_messages.spec.js should pass on main. The late day counts in the application should match what the tests expect.

**To Reproduce**
1. Run the Cypress (Gradeable) CI suite on the current main branch
2. Observe 4 failures in late_submission_warning_messages.spec.js
3. Failing tests:
"Confirmation for the first submission with 2 remaining late days and 1 extension"
"Warning message for the second submission with 0 valid remaining late day"
"Confirmation for the first submission with 2 remaining late days and 1 extension for teams"
"Warning message for the second submission with one team member having 0 remaining late days"
4. See error : late day counts are off by one, likely due to date/time sensitivity

**Configuration**
CI environment (GitHub Actions, Ubuntu)
Browser: Chrome 146 (headless)

**Screenshots**
N/A — failures visible in CI logs.

**Additional context**
This appears to be a date/time sensitivity issue similar to the cascading failures seen in PR #12571 where date changes affected test behavior. 

CI run showing the failure on main: https://github.com/Submitty/Submitty/actions/runs/23856035445

### PR DESCRIPTION
**Pull request by GarvitKhandelwal31**
### Why is this Change Important & Necessary?
Fixes #12732 
This change targets deterministic test setup/synchronization issues in the spec itself (date formatting consistency, save-status timing, and selector reliability), which were causing intermittent and cascading failures.

### What is the New Behavior?
Root cause of flaky dates was that  instead of using the machine's local timezone (via new Date().toISOString()), which varies depending on where the runner is running relative to the Submitty server, you now format dates strictly in the America/New_York timezone matching the server.

### What steps should a reviewer take to reproduce or test the bug or new feature?
The CI/Gradeable test should pass.

### Other information
- Breaking change: No.

### CODE REVIEW

**Comment by GarvitKhandelwal31** [comment]
It seems the issue is still not resolved. Will look at it tomorrow.

File: site/cypress/e2e/Cypress-Gradeable/late_submission_warning_messages.spec.js
Comment (IDzyre) [inline site/cypress/e2e/Cypress-Gradeable/late_submission_warning_messages.spec.js:1]: The "failure description" in the Ci runners is a bit misleading. While it does fail at the "saved changes" message, that is not why this test is failing at first. CI re-runs the tests, and since we don't fully reset state before each test, all previous actions are kept, so any changes to gradeable dates are kept. 

Adding more waiting will not make the test pass more often, it will just take longer to fail. On this branch, when it fails, it takes more than 6 minutes, on main when it passes it takes ~2 minutes.

If you want to see why the test is failing, you can download and watch the video of the test running by going to the `Summary` page of the CI, and scrolling down to the bottom in the `Artifacts` section. There you can download a screenshot of the time it failed, and the whole video of the test.

**Comment by GarvitKhandelwal31** [comment]
@IDzyre The current stabalization was passing 8/11 checks locally. In that so far  I have added timezone-aware helpers so the test no longer depends on CI runner clock drifting from Submitty server timezone.
Replaced save status check with small settle helper, so that autosave had time to finish before next step. Some other changes were also done.
The current state- It is failing on non team 1 allowed/ 1 remaining late days, 2 remaining late days+ extension in both team and non team state
In all of these cases the confirm dialog is resolving to 20533 days late, while submission page itself shows realistic late-day values after submit completes.
I would look into them shortly.

**Comment by GarvitKhandelwal31** [comment]
It's actually good that late submission Cypress was passing. Can you re-run the workflow again so that we could confirm if the fix is deterministic.

Moreover it's strange why the other workflows are failing. I don't find any reason for docker_ui spec to fail? What is your opinion on it @IDzyre?

**Comment by IDzyre** [comment]
> It's actually good that late submission Cypress was passing. Can you re-run the workflow again so that we could confirm if the fix is deterministic.
> 
> Moreover it's strange why the other workflows are failing. I don't find any reason for docker_ui spec to fail? What is your opinion on it @IDzyre?

Don't worry about docker_ui, that one fails from time to time for network issues.

**Comment by GarvitKhandelwal31** [comment]
> > It's actually good that late submission Cypress was passing. Can you re-run the workflow again so that we could confirm if the fix is deterministic.
> > Moreover it's strange why the other workflows are failing. I don't find any reason for docker_ui spec to fail? What is your opinion on it @IDzyre?
> 
> Don't worry about docker_ui, that one fails from time to time for network issues.

Also I saw that gradeable upload spec was also failing on many of the PRs where it was completed unrelated. I tried running that on main branch. Failed locally. Should I open another PR for it?

**Comment by IDzyre** [comment]
If you want to, sure.

**Comment by GarvitKhandelwal31** [comment]
Anything further you would require from my side before the the final approval?

File: site/cypress/e2e/Cypress-Gradeable/late_submission_warning_messages.spec.js [hunk 1]
```diff
@@ -170,12 +189,19 @@ const SubmitAndCheckMessage = (gradeable_type, upload_file1, invalid_late_day, v
     cy.logout();
 };
 
+const waitForSaveStatusSettled = () => {
+    // Give autosave debounce a chance to fire before checking final state.
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(300);
+    cy.get('[data-testid=save-status]', { timeout: 30000 }).should('not.contain', 'Saving');
+};
+
```
Comment (IDzyre) [inline site/cypress/e2e/Cypress-Gradeable/late_submission_warning_messages.spec.js:195]: Can you explain the rational behind changing from waiting for the all changes saved to this?

Comment (GarvitKhandelwal31) [reply site/cypress/e2e/Cypress-Gradeable/late_submission_warning_messages.spec.js:195]: My rationale for changing that assertion was to avoid a flaky timing edge where the status briefly transitions through Saving.../Saving Changes and the strict text check fires at the wrong moment in CI, especially during retries. The intent was to wait for autosave to settle without depending on one exact transient UI string.

Comment (IDzyre) [reply site/cypress/e2e/Cypress-Gradeable/late_submission_warning_messages.spec.js:195]: I wouldn't focus on retries in CI, because we do not currently reset the state of the gradeable before each test properly. The tests will fail on retry in probably 90% of the tests because the gradeable already has the edits done, so it won't have any changes to make. I would argue that is a good thing. If the test requires a retry, that means something wasn't good in the beginning, and we should catch that and fix it, instead of just trying to get the tests to pass. 

The save should happen within a second, maybe up to 10 if something is going on with the machine. I don't think this is a 'flaky edge case' for the saving changes, watching the vidoes the saving changes happens within a couple of seconds, unless something failed and it is retried. We should fix the part that caused it to be retried. I don't think we should be adding more wait time to an already long test. Worst case scenario for this test I believe is something greater than six minutes. 

Also, this doesn't really fix the issue if it was a 'flaky edge case', watching the videos it says `Expected "All Changes Saved" found ""`, so `""` does not contain `"Saving"`, so I don't see how that would help in that case.

**Comment by GarvitKhandelwal31** [comment]
Closing in favor of #12904

**Comment by dagemcn** [comment]
Apologies, I did not see you were working on this before I opened my PR.

**Comment by GarvitKhandelwal31** [comment]
> Apologies, I did not see you were working on this before I opened my PR.

Na don't worry on that, anyways I was stuck at this PR
